### PR TITLE
Update protein-translation tests

### LINKED ISSUE #1103: protein-translation: update tests
As identified in #784, some of the exercise test-suites are out-of-date, protein-translation is one of them. It would therefore be really helpful for people to go through and update the test suite to match the canonical data.

Tests should typically be ordered according to their order in the canonical data, since this makes it easier to check and update them in future. If you think that there's a better order, please let us know your reasoning in your PR.

If there are some additional tests that currently exist in this track but which aren't in the canonical data, these can be kept if you think that they are useful. They should be placed at the bottom of the test cases, and it would also be super-helpful if you could add a comment before them saying # Additional tests for this track to make it clear that they aren't part of the canonical data.

The version string should be placed just before the first test class and followed by a blank line. For this exercise, the version string is:

`# Tests adapted from problem-specifications/canonical-data.json @ v1.0.0`

**Test suite:** [/exercism/python/exercises/protein-translation/protein_translation_test.py](https://github.com/exercism/python/blob/master/exercises/protein-translation/protein_translation_test.py)
**Canonical data:** [exercism/problem-specifications//exercises/protein-translation/canonical-data.json](https://github.com/exercism/problem-specifications/blob/master/exercises/protein-translation/canonical-data.json)

### PR DESCRIPTION
**Pull request by sjwarner-bp**
Fixes #1103

### CODE REVIEW

**Comment by sjwarner-bp** [comment]
I've just seen this error - can someone let me know if there is a maximum line size, as this seems to be the error (but I'd like to check first) :slightly_smiling_face:

**Comment by N-Parsons** [comment]
Hi @sjwarner-bp, if you click through to Travis-CI using the "details" link in the bottom box, it will take you to an overview of the tests. You can then pick one of the Python versions (usually doesn't matter which) and then it will show an output of what's wrong.

In your case, it looks like `flake8` formatting errors ([link](https://travis-ci.org/exercism/python/jobs/304692531):
```shell
$ flake8
./exercises/protein-translation/protein_translation_test.py:7:1: E302 expected 2 blank lines, found 1
./exercises/protein-translation/protein_translation_test.py:49:80: E501 line too long (80 > 79 characters)
./exercises/protein-translation/protein_translation_test.py:54:80: E501 line too long (82 > 79 characters)
./exercises/protein-translation/protein_translation_test.py:59:80: E501 line too long (83 > 79 characters)
./exercises/protein-translation/protein_translation_test.py:61:33: E231 missing whitespace after ','
./exercises/protein-translation/protein_translation_test.py:61:44: E231 missing whitespace after ','
```

You can also install `flake8` locally using `pip install flake8`, and then run it with `flake <path>`.

**Comment by cmccandless** [comment]
@N-Parsons  (cc @sjwarner-bp )
>You can also install `flake8` locally using `pip install flake8`, and then run it with `flake8 [path]`.

I think you accidentally dropped an "8". Also note that path is optional and defaults to "./" 😄

**Comment by sjwarner-bp** [comment]
Perfect - thanks for letting me know guys, I've installed it locally and will remember to use it in future! :slightly_smiling_face:

File: exercises/protein-translation/protein_translation_test.py [hunk 1]
```diff
@@ -35,25 +37,34 @@ def test_identifies_stop_codons(self):
         for codon in ['UAA', 'UAG', 'UGA']:
             self.assertEqual('STOP', of_codon(codon))
```
Comment (cmccandless) [inline exercises/protein-translation/protein_translation_test.py]: Per [discussion](https://github.com/exercism/problem-specifications/pull/997/commits/4c5b8e8bf2ee851b1a13e771c7e782ba4b972a78#r149712640) in exercism/problem-specifications#997, I don't think we should test for intermediate results. Thus, all tests should test `of_rna()` (`proteins()` in the spec).

File: exercises/protein-translation/protein_translation_test.py [hunk 2]
```diff
@@ -35,25 +37,34 @@ def test_identifies_stop_codons(self):
...
 
-    def test_translates_rna_strand_into_correct_protein(self):
+    def test_translates_rna_strand_into_correct_protein_list(self):
         strand = 'AUGUUUUGG'
         expected = ['Methionine', 'Phenylalanine', 'Tryptophan']
         self.assertEqual(expected, of_rna(strand))
```
Comment (cmccandless) [inline exercises/protein-translation/protein_translation_test.py]: In an effort to maintain consistency with the spec, `of_rna()` should probably be renamed to `proteins()`.

**Comment by sjwarner-bp** [comment]
Hi @cmccandless - just to double check so I don't make the wrong edits, all `of_codon()` and `of_rna()` should be renamed to `proteins()`?

**Comment by cmccandless** [comment]
@sjwarner-bp Correct, however note that the tests that currently use `of_codon()` will need to be modified to expect a list (albeit with only 1 element each).

**Comment by sjwarner-bp** [comment]
Perfect, I understand you :smile:

**Comment by sjwarner-bp** [comment]
Sorry, I have just picked this up again after a hectic week! I'm trying to do this but running into trouble.

Just to clarify, the things I need to do to satisfy this is:
* Change all reference of `of_codon` and `of_strand` to `protein`
* Update the skeleton file with a single definition for `protein` and nothing else
* Update the reference solution

So I have done the first two, but the final one is causing me a little confusion - do you have any guidance?

**Comment by cmccandless** [comment]
@sjwarner-bp Have you tried simply renaming (in `example.py`) `of_rna`->`protein`, with no other changes?

**Comment by sjwarner-bp** [comment]
Hi @cmccandless , I have renamed `of_rna` to `protein` and I end up with 13 failed tests, all of which have `AssertationError <some_output> != None`

**Comment by cmccandless** [comment]
@sjwarner-bp Can you please drop the contents of your `example.py` and `protein_translation_test.py` in a multi-file [gist](https://gist.github.com/) so I can take a look?

**Comment by sjwarner-bp** [comment]
Sure thing - [here it is](https://gist.github.com/sjwarner-bp/8dd5a0becad35150898a4840ff68c888)

**Comment by cmccandless** [comment]
Don't forget to change
```Python
from protein_translation import protein
```
to
```Python
from example import protein
```
when running tests locally (revert before committing).

**Comment by sjwarner-bp** [comment]
Yup, done that - I still end up with 8 failing tests for some reason though :scream:. I've tried the (previously) `of_codon` with and without list notation around the parameter.

**Comment by cmccandless** [comment]
Make sure the tests that were previously `of_codon` tests both pass in *and* expect a list.

**Comment by sjwarner-bp** [comment]
Do you mean something like the following?

```
def test_AUG_translates_to_methionine(self):
        self.assertEqual(['Methionine'], protein(['AUG']))
```

I've tried this (and then without the `[ ]`) but it doesn't seem to work.

**Comment by cmccandless** [comment]
Sorry; I wasn't paying full attention to this and I didn't take the time to fully understand the inputs/outputs.

According to the [canonical data](https://github.com/exercism/problem-specifications/blob/8793a059fe18618fffeb97630baf76eebd5bf9a8/exercises/protein-translation/canonical-data.json), here are a few notes:
- Tests should only reference `proteins()` (I told you to call it `protein`, but should pluralize for consistency)
- `proteins()` expected a *string* (my bad, told you to provide a list, but that's wrong)
- `proteins()` returns a list, regardless of the number of elements in the input
- If `proteins()` encounters a STOP codon, translation ceases and the results so far are returned. The STOP protein itself is *not* included.

See if any of that helps. I was able to make a few quick edits to your code to make it work, but you will benefit more from this if you figure it out yourself :smile:

File: exercises/protein-translation/protein_translation_test.py [hunk 3]
```diff
@@ -1,58 +1,69 @@
 import unittest
 
-from protein_translation import of_codon, of_rna
+from protein_translation import proteins
+
+# Tests adapted from problem-specifications/canonical-data.json @ v1.0.0
```
Comment (cmccandless) [inline exercises/protein-translation/protein_translation_test.py]: For consistency with other exercises, can you please change the spacing around this line? Should be 2 blank lines above, 1 below. (In the back of my mind I have a feeling I may have already said this, but switched the numbers; if so, I apologize for the misinformation)

**Comment by cmccandless** [comment]
@sjwarner-bp Merged. Thanks for working on this!

**Comment by sjwarner-bp** [comment]
Thanks for helping me out!

**Pull request merged**